ca/ca.go

```diff
@@ -104,6 +147,7 @@ func NewCertificateAuthorityImpl(
 	pa core.PolicyAuthority,
 	boulderIssuers []*issuance.Issuer,
 	certificateProfile *issuance.Profile,
+	certificateProfiles []*issuance.Profile,
```

Comment:
Let's not support having both of these populated. Instead, the //ca package should only take `certificateProfiles map[string]*issuance.Profile` as input, and it should be up to //cmd/boulder-ca/main.go to (if only a single Profile was configured) convert it into such a map before calling this constructor.